src/errors.rs

```diff
@@ -37,7 +38,7 @@ pub enum DeployError {
 
     /// An error occured encoding deployment parameters with the contract ABI.
     #[error("error ABI ecoding deployment parameters: {0}")]
-    Abi(#[from] AbiError),
+    Abi(AbiErrorKind),
```

Comment:
So the problem was that AbiError itself is not Sync?

Reply:
Correct.